app/code/community/Bolt/Boltpay/Helper/Data.php

```diff
@@ -89,4 +89,25 @@ public function canUseForCountry($country)
 
         return true;
     }
+
+    /**
+     * Resets rounding deltas before calling collect totals which fixes bug in collectTotals that causes rounding errors
+     * when a percentage discount is applied to a quote
+     *
+     * @param Mage_Sales_Model_Quote $quote
+     * $param boolean $clearTotalsCollectedFlag Determines whether to set setTotalsCollectedFlag to false
+     *
+     * @return Mage_Sales_Model_Quote
+     */
+    public function collectTotals($quote, $clearTotalsCollectedFlag = false) {
```

Comment:
@johnmccombs1 can you write unit test for this? (please push back if it'd be too complicated)

Reply:
I'll be creating a regression test for this in the future, but in order to expedite this I'm going to go ahead and migrate this.